gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java

```diff
@@ -454,6 +515,29 @@ private List<RCMRMT030101UKEhrComposition> getCompositionsContainingLinkSets(RCM
             .toList();
     }
 
+    protected Optional<RCMRMT030101UKObservationStatement> getObservationStatementByCodeableConceptCode(
+        RCMRMT030101UK04EhrExtract ehrExtract,
+        RCMRMT030101UKObservationStatement referencedObservationStatement) {
+
+        List<RCMRMT030101UKObservationStatement> observationStatements = ehrExtract.getComponent().stream()
+            .map(RCMRMT030101UKComponent::getEhrFolder)
+            .map(RCMRMT030101UKEhrFolder::getComponent)
+            .flatMap(Collection::stream)
+            .map(RCMRMT030101UKComponent3::getEhrComposition)
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .flatMap(CompoundStatementResourceExtractors::extractAllObservationStatements)
+            .toList();
```

Comment:
Does this code satisfy this condition?

```
Be a child of the same EhrComposition
```

Reply:
Also the condition:
```
The replaced ObservationStatement must have a single pertinentInformation / pertinentAnnotation which ends in ...
```

Reply:
I think I can see that final condition being met on line 253, but the downside of this approach, is that the check happens after the `findFirst` call which means that we could be missing the `ObservationStatement` if the actual match isn't the first.

Reply:
I think we covered it with a test when the ordering is different and it works OK.

Reply:
> Also the condition:
> 
> ```
> The replaced ObservationStatement must have a single pertinentInformation / pertinentAnnotation which ends in ...
> ```

Yes, we've got it confirmed with tests.